Protecs/src/main/java/uk/ac/ucl/protecs/sim/WorldBankCovid19Sim.java

```diff
@@ -193,14 +193,25 @@ public void step(SimState arg0) {
 		
 
 		if (this.demography) {
-			schedule.scheduleRepeating(Demography.CreateBirths(this), this.param_schedule_updating_locations, params.ticks_per_day);
-
-			// SCHEDULE CHECKS ON MORTALITY AND LOGGING
-			schedule.scheduleRepeating(Demography.ReportBirthRates(this), this.param_schedule_reporting, params.ticks_per_day);
-			schedule.scheduleRepeating(Demography.CheckMortality(this), this.param_schedule_reporting, params.ticks_per_day);
+			Demography myDemography = new Demography();
+			for(Person a: agents) {
+				// Trigger the aging process for this person
+				Demography.Aging agentAging = myDemography.new Aging(a, params.ticks_per_day);
+				schedule.scheduleOnce(a.getBirthday()*params.ticks_per_day, this.param_schedule_reporting, agentAging);
+				// Trigger the process to determine mortality each year
+				Demography.Mortality agentMortality = myDemography.new Mortality(a, params.ticks_per_day, this);
+				schedule.scheduleOnce(0, this.param_schedule_reporting, agentMortality);
```

Comment:
would it be possible to set mortality and birth-giving to start from the agent's birthday too?

Reply:
It would be, but can I ask why? Is it do make sure there isn't a big stack of these events all happening at once?

Reply:
(per conversation, this is not a concern and handled otherwise)